docs/deploys.md

```diff
@@ -0,0 +1,120 @@
+
+Halyard handles this deployment by installing all Spinnaker service git repos
+in a user-configurable directory, with a user-configurable `upstream` and
+`origin`.  There will be a special version called `HEAD` that supersedes any 
+configuration in our profile registry that will take all services at git `HEAD` 
+to use for config generation.
+
+### Flotilla Deployments
+
+> Timeline: Varies by cloud provider. See below.
+
+1. **Configuration**: Varies by cloud provider.
+2. **Service Dependency Setup**: Varies by cloud provider.
+3. **Deployent & Updates**: This deployment involves first deploying a separate 
+ Orca, Clouddriver and Redis using their published `-bootstrap.yml` profiles to 
+ only communicate with one-another, and then orchestrate the remainder of the 
+ update/deployment through them by publishing red/black update pipelines that 
+ redeploy desired services with a cluster history size of 2 for fast rollbacks.
+ There are a few nuances here:
+  * **Orca**: Orca must have an unlimited cluster history size, and Halyard
+   must decide when it is safe to kill an Orca node.
+  * **Cache-schema changes**: Cache changes have broken service updates
+   the past, so we need to handle these separately. Since there will be window 
+   when instances from both version with incompatable caches schemas will 
+   share the same Redis cache, flushing the cache is not an option. The only 
+   alternative is to scale down the old version to 0 instances, and the flush
+   the redis cache where applicable. The critical thing to note here is that
+   the only way Halyard can know to do that is when a services Major version
+   increases.
+
+#### Kubernetes 
+
+> Timeline: End of March. The core features are in, but a few touchups as
+> detailed in https://github.com/spinnaker/halyard/projects/1
+
+1. **Configuration**: Halyard publishes all configuration as Kubernetes
+ secrets, and describes to the individual services where to mount them.
+2. **Service Dependency Setup**: The only external dependency is redis, and if
+ it is not provided, Spinnaker will deploy a Redis instance.
+3. **Deployent & Updates**: See above.
+
+#### GCE
+
+> Timeline: Middle April.
+
+1. **Configuration**: Halyard uses the configured Vault storage and access 
+ token to write all configuration as secrets per-instance, which the bootscript
+ of the instance will download and place in their intended directories. The
+ secret format looks roughly like this:
+ ```
+ /secret/spinnaker/clouddriver-{id}: {"configs": ["config1, "config2"] }
+ /secret/spinnaker/clouddriver-{id}/config1: {"path": "/a/b", "contents" "xyz"}
+ /secret/spinnaker/clouddriver-{id}/config2: {"path": "/z/y", "contents" "cba"}
+ ```
+ The instance will be provided with top-level secret id `clouddriver-{id}` and
+ use that to download the secrets `config1` & `config1`, and place their
+ `contents` into `path`.  Service endpoints will be configured using typical Consul DNS names, which
```

Comment:
The problem with the ILBs is that they don't support internal DNS. Until they do, any changes to your endpoint configuration will require redeploying all services to get new endpoint info.